win32/select.c

```diff
@@ -135,15 +135,23 @@ PHPAPI int php_select(php_socket_t max_fd, fd_set *rfds, fd_set *wfds, fd_set *e
 				for (i = 0; i < n_handles; i++) {
 					if (WAIT_OBJECT_0 == WaitForSingleObject(handles[i], 0)) {
 						if (SAFE_FD_ISSET(handle_slot_to_fd[i], rfds)) {
-							FD_SET((uint32_t)handle_slot_to_fd[i], &aread);
+							DWORD avail_read = 0;
+							if (GetFileType(handles[i]) != FILE_TYPE_PIPE
+								|| !PeekNamedPipe(handles[i], NULL, 0, NULL, &avail_read, NULL)
```

Comment:
I'm not sure I understand this part of condition. Does it add the handler as ready it if the `PeekNamedPipe` fails (return 0)?

Reply:
Yes. Only if the handle refers to a pipe, `PeekNamedPipe()` succeeds, but there is no data available, the handle is *not* added to the set. If the pipe has been closed ("broken pipe"), users may still need to read to detect EOF.

Reply:
Is that also how it works with sockets? I mean if the connection get closed, does it report read ready?